tests/acceptance/01_vars/02_functions/not.cf

```diff
@@ -0,0 +1,98 @@
+######################################################
+#
+#  Test that not() behaves as expected
+#
+#####################################################
+body common control
+{
+    inputs => { "../../default.cf.sub" };
+    bundlesequence  => { default("$(this.promise_filename)") };
+    version => "1.0";
+}
+
+#######################################################
+
+bundle agent init
+{
+}
+
+#######################################################
+
+bundle agent test
+{
+  meta:
+    "description" -> { "CFE-3470" }
+      string => "Test that not() behaves as expected";
+
+  classes:
+    "ok"
+      scope => "namespace",
+      and => {
+        # Simple case:
+        "any",
+        not("!any"),
+        "cfengine",
+        not("!cfengine"),
+
+        # Variable expansion:
+        "$(true_variable)",
+        not("$(false_variable)"),
+        not(not("$(true_variable)")),
+        not(not(not("$(false_variable)"))),
```

Comment:
:exploding_head:

Reply:
It's easy if you read the previous line, for every not you add, alternate true vs false ;)